controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/InternalDeploymentTester.java

```diff
@@ -137,26 +105,41 @@ public InternalDeploymentTester() {
         domain.services.put(ATHENZ_SERVICE, new AthenzDbMock.Service(true));
     }
 
+    public ReadyJobsTrigger readyJobsTrigger() {
+        return readyJobsTrigger;
+    }
+
+    public NameServiceDispatcher nameServiceDispatcher() {
+        return nameServiceDispatcher;
+    }
+
+    /** Returns the default deployment context owned by this */
+    public DeploymentContext deploymentContext() {
+        return defaultContext;
+    }
+
+    /** Create a new deployment context for given application */
+    public DeploymentContext newDeploymentContext(String tenantName, String applicationName, String instanceName) {
+        return newDeploymentContext(ApplicationId.from(tenantName, applicationName, instanceName));
+    }
+
+    /** Create a new deployment context for given application */
+    public DeploymentContext newDeploymentContext(ApplicationId instance) {
+        return new DeploymentContext(instance, this);
+    }
+
     /** Create a new application with given tenant and application name */
     public Application createApplication(String tenantName, String applicationName, String instanceName) {
-        return tester.controllerTester().createApplication(tester.controllerTester().createTenant(tenantName,
-                                                                                                  athenzDomain + nextDomainId.getAndIncrement(),
-                                                                                                  nextPropertyId.getAndIncrement()),
-                                                           applicationName,
-                                                           instanceName,
-                                                           nextProjectId.getAndIncrement());
+        return newDeploymentContext(tenantName, applicationName, instanceName).application();
     }
 
     /** Submits a new application, and returns the version of the new submission. */
-    public ApplicationVersion newSubmission(TenantAndApplicationId id, ApplicationPackage applicationPackage,
-                                            SourceRevision revision, String authorEmail, long projectId) {
-        return jobs.submit(id, revision, authorEmail, projectId, applicationPackage, new byte[0]);
+    public ApplicationVersion newSubmission(TenantAndApplicationId id, ApplicationPackage applicationPackage, SourceRevision sourceRevision) {
```

Comment:
We should probably inline methods like this? (Work for later.)

Reply:
I think these methods should be removed altogether, and rather have tests always use a context.

Reply:
Exactly.